arangod/Aql/OptimizerRulesJoin.cpp

```diff
@@ -344,8 +344,8 @@ bool checkCandidatesEligible(
       auto* lhs = root->getMember(0);
       auto* rhs = root->getMember(1);
 
-      if (!joinConditionMatches(plan, lhs, rhs, c, candidates[i - 1]) &&
-          !joinConditionMatches(plan, lhs, rhs, candidates[i - 1], c)) {
+      if (!joinConditionMatches(plan, lhs, rhs, c, candidates[0]) &&
+          !joinConditionMatches(plan, lhs, rhs, candidates[0], c)) {
```

Comment:
This change will cause a behavior change:
Instead of comparing candidate[1] with candiate[0], and then candidate[2] with candidate[1], ... we will now compare candiate[1] with candidate[0], then candidate[2] with candidate[0] etc.
Not saying this is wrong in any way, but the behavior now can be different if there are more than 2 candidates.

Reply:
will be generalized in a follow up PR